docs/ECO-INIT.md

```diff
@@ -0,0 +1,153 @@
+# ECO Initialization
+
+This document describes the ECO initialization process, specifically how
+an ECO goes from "configured in the controller" to "launched and running".
+This includes:
+
+* configuration creation
+* image download and verification
+* network creation and initialization
+* ECO launch
+
+For a deeper understanding off ECOs, please read [ECOs.md](./ECOs.md).
+
+## Process
+
+The high-level process for preparing an ECO follows this flow. In each section, the
+responsible component is placed in brackets. All communication between components,
+i.e. configs and statuses, are performed using publish-subscribe via
+[pubsub](./IPC.md).
+
+1. Create App: An ECO is configured on a valid controller, e.g. [zedcloud](https://zededa.com)
+1. Receive Device Config ([zedagent](../pkg/pillar/cmd/zedagent)): EVE device receives and [processes](../pkg/pillar/cmd/zedagent/handleconfig.go#L412) an updated device config message which contains the new ECO app
+1. Create Local App Config ([zedagent](../pkg/pillar/cmd/zedagent)): [parse the app config](../pkg/pillar/cmd/zedagent/parseconfig.go#L453) in the device config message to [create and publish](../pkg/pillar/cmd/zedagent/parseconfig.go#L491-558) an [AppInstanceConfig](../pkg/pillar/types/zedmanagertypes.go#L46) for the ECO
```

Comment:
Referencing line numbers means the documentation is probably already out of date. Is it possible to reference function names instead?

Reply:
Unfortunately not; GitHub does not support it. I had _precisely_ the same thought (they say great minds think alike). Any ideas on how to fix this?

Reply:
You _can_ link to a specific revision of a file, so that the line number always will point to that function, but then the file itself may be out of date.